tests/close_events.rs

```diff
@@ -0,0 +1,268 @@
+#[macro_use]
+mod util;
+
+use mio::event::{Events, Source};
+use mio::net::{TcpListener, TcpStream};
+use mio::{Interests, Poll, Token};
+use std::io::{Read, Result, Write};
+use std::net::{Shutdown, SocketAddr};
+use std::sync::{mpsc, Arc, Barrier};
+use std::thread::{self, JoinHandle};
+use std::time::Duration;
+use util::{any_local_address, any_local_ipv6_address, expect_events, init_with_poll, ExpectEvent};
+
+const DATA: &[u8] = b"hello";
+const TOK: Token = Token(0);
+
+macro_rules! generate_tests {
```

Comment:
Do we really need a macro for this? Can't we use the same strategy we use for the other net tests, e.g: https://github.com/tokio-rs/mio/blob/162cdd3868742ed494ba03e7c843708eebd6849a/tests/tcp_listener.rs#L28-L37

Reply:
Yea I initially took this approach and had some issues with errors getting buried by the call through `smoke_test_tcp_listener`. I may have had it set up wrong I can try changing it back.